### PR TITLE
feat(fetch): make fetch api public

### PR DESCRIPTION
**Pull request by yury-s**
#1774 
#5999

### CODE REVIEW

File: docs/src/api/class-fetchresponse.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+This method will throw if the response body is not parsable via `JSON.parse`.
+
+## async method: FetchResponse.json
+* langs: csharp
+- returns: <[null]|[JsonElement]>
+
+Returns the JSON representation of response body.
+
+This method will throw if the response body is not parsable via `JSON.parse`.
```
Comment (dgozman) [inline docs/src/api/class-fetchresponse.md:49]: Is "JSON.parse" a thing in dotnet?

File: docs/src/api/class-route.md [hunk 2]
```diff
@@ -220,6 +220,11 @@ Optional response body as raw bytes.
 File path to respond with. The content type will be inferred from file extension. If `path` is a relative path, then it
 is resolved relative to the current working directory.
 
+### option: Route.fulfill.response
+- `response` <[FetchResponse]>
+
+[FetchResponse] to fulfill route's request with. Individual fields of the response (such as headers) can be overridden using fulfill options.
```
Comment (dgozman) [inline docs/src/api/class-route.md:226]: nit: "using other fulfill options"

File: types/types.d.ts [hunk 3]
```diff
@@ -1995,6 +1995,34 @@ export interface Page {
    */
   exposeFunction(name: string, callback: Function): Promise<void>;
 
+  /**
+   * Sends HTTP(S) request and returns its response. The method will populate request cookies from the context and update
+   * context cookies from the response. The method will automatically follow redirects.
+   * @param urlOrRequest Target URL or Request to get all fetch parameters from.
+   * @param options
+   */
+  fetch(urlOrRequest: string|Request, options?: {
```
Comment (dgozman) [inline types/types.d.ts:2004]: So, you actually allow passing Request and options together?

**Review by dgozman — APPROVED** [review]

File: docs/src/api/class-fetchresponse.md (hunk 1 above)
Comment (yury-s) [reply docs/src/api/class-fetchresponse.md:49]: Dunno, I copied it from our current Response docs. @avodovnik ^^

File: types/types.d.ts (hunk 3 above)
Comment (yury-s) [reply types/types.d.ts:2004]: As discussed off this PR let's use same options for all overloads.

**Pull request merged**